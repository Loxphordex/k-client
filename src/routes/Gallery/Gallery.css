.generated-images {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-flow: column;
  margin-top: 24px;
  margin-bottom: 8vh;
}

.img-container {
  position: relative;
}

.img-container img {
  width: 100%;
  max-width: 420px;
  display: block;
  margin-bottom: 4%;
  border: 1px solid #2c2e36;
  border-radius: 10px;
}

.img-container h2 {
  text-align: center;
  font-size: 0.8em;
  margin-top: 0;
  margin-bottom: 18px;
}

@media only screen and (min-width: 760px) {
  .gallery-area {
    margin-top: 30px;
  }
  .generated-images {
    flex-flow: wrap;
    justify-content: flex-start;
    align-items: flex-end;
    max-width: 1200px;
    margin: auto;
    margin-bottom: 15vh;
  }

  .img-container {
    box-sizing: border-box;
    display: grid;
    grid-template-rows: 0.5fr 0.5fr;
    position: relative;
    max-width: 300px;
    margin: 24px;
  }

  .img-container img {
    grid-row: 1/3;
    width: 250px;
    margin-bottom: 0;
  }

  .img-container img:hover {
    opacity: 0.65;
  }

  .img-container h2 {
    margin-top: 4px;
  }
}