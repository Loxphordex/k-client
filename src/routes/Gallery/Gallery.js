import React from 'react'
import config from '../../config'
import { CloudinaryContext } from 'cloudinary-react'

// COMPONENTS
import GenerateImages from '../../components/GenerateImages/GenerateImages'

export default class Gallery extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      index: 1,
      images: [],
      allImages: [],
    }
  }

  render() {
    const { images } = this.state
    return(
      <section className='gallery-area'>
        <CloudinaryContext cloudName={config.CLOUD_NAME} className='cloud-context'>
          <GenerateImages images={images} />
        </CloudinaryContext>
      </section>
    )
  }
}