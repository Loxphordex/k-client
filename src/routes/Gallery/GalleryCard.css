.img-container h2:hover {
    text-decoration: underline;
}

@media only screen and (min-width: 760px) {
  .img-container:hover .hover-background {
    width: 280px;
    height: 380px;
    z-index: 800;
    display: block;
    position: absolute;
    top: -6px;
    left: -10px;
    box-shadow: 2px 20px 46px 6px #10101345;
  }

  .img-container:hover img {
    width: 280px;
    z-index: 1000;
    left: -10px;
    right: 0px;
    top: -6px;
    box-shadow: none;
  }  

  .img-container:hover h2 {
    z-index: 900;
    right: 0px;
    bottom: -36px;
    font-size: 0.9em;
  }

  .img-container:hover .gallery-type-header {
    opacity: 1;
    z-index: 900;
    bottom: -34px;
  }

  .img-container:hover .gallery-image-price {
    opacity: 1;
    z-index: 900;
    bottom: -92px;
  }
}