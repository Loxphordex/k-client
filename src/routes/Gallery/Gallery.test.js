import React from 'react'
import ReactDOM from 'react-dom'
import Gallery from './Gallery'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Gallery', () => {
  it('renders the complete page', () => {
    const wrapper = shallow(<Gallery />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Gallery />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})