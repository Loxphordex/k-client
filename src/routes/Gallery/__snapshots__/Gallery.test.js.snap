// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders the complete page 1`] = `
<section
  className="gallery-area"
>
  <CloudinaryContext
    className="cloud-context"
    cloudName="dws2woreg"
    includeOwnBody={false}
  >
    <GenerateImages
      images={Array []}
      setDeleteId={[Function]}
      setEditorImageId={[Function]}
    />
  </CloudinaryContext>
</section>
`;
